src/lib/dapp/DAppSigner.ts

```diff
@@ -32,18 +33,53 @@ import {
 } from '@hashgraph/sdk'
 import type { ISignClient } from '@walletconnect/types'
 
-import { ledgerIdToCAIPChainId } from '../shared'
+import {
+  ExecuteTransactionResult,
+  HederaJsonRpcMethod,
+  SignAndExecuteQueryResult,
+  SignAndExecuteTransactionResult,
+  SignMessageResult,
+  SignTransactionResult,
+  base64StringToSignatureMap,
+  ledgerIdToCAIPChainId,
+  transactionBodyToBase64String,
+  transactionToBase64String,
+  transactionToTransactionBody,
+} from '../shared'
 
 export class DAppSigner implements Signer {
+  static signClient: ISignClient
+  public signerAccountId: string
+
+  /**
+   * These are the node accounts that will be utilized for signing transactions.
+   */
+  public nodesAccountIds: AccountId[] = []
+
   constructor(
     private readonly accountId: AccountId,
-    private readonly signClient: ISignClient,
     public readonly topic: string,
     private readonly ledgerId: LedgerId = LedgerId.MAINNET,
-  ) {}
+  ) {
+    this.signerAccountId = `${ledgerIdToCAIPChainId(ledgerId)}:${accountId.toString()}`
+    this.nodesAccountIds = [AccountId.fromString('0.0.3')]
+  }
+
+  static initialize(signClient: ISignClient) {
```

Comment:
I think it was more elegant to pass the signClient in the constructor like it was before instead of using a static method to set a static prop that's shared between all instances.

Reply:
That is debatable. For now, we have nothing to ease the dApp developers' integration. Let's start with this contribution and then discuss minor details like this.